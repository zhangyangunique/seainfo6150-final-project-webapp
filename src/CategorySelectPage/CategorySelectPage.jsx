import React from "react";
import { Link } from "react-router-dom";
import styles from "./CategorySelectPage.module.css";

const CategorySelectPage = () => {
	return (
		<div className={styles.container}>
			<h2>Please select the topic you are interested in</h2>
			<div className={styles.categoryCard}>
				<Link to="/category/sport">
					<p>Sport</p>
				</Link>
				<Link to="/category/food">
					<p>Food</p>
				</Link>
				<Link to="/category/history">
					<p>History</p>
				</Link>
				<Link to="/category/tourism">
					<p>Tourism</p>
				</Link>
			</div>
		</div>
	);
};

export default CategorySelectPage;
