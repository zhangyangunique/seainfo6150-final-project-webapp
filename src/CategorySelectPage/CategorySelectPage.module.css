.container {
    margin: 0 auto;
    text-align: center;
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.categoryCard {
    margin: 0 auto;
    width: 50%;
    min-width: 300px;
    margin-bottom: 100px;
}

.categoryCard a {
    display: flex;
    align-items: center;
    text-decoration: none;
    border: 2px solid black;
    border-radius: 100px;
    margin-top: 50px;
    padding: 50px;
}

.categoryCard a p {
    margin: 0 auto;
    font-size: 60px;
}

@media screen and (min-width: 525px) {
    .categoryCard a p {
        font-size: 50px;
    }
}

@media screen and (min-width: 800px) {
    .categoryCard a p {
        font-size: 40px;
    }
}

@media screen and (min-width: 1200px) {
    .categoryCard a p {
        font-size: 30px;
    }
}