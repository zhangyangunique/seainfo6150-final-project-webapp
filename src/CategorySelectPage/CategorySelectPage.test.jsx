import React from "react";
import { BrowserRouter } from 'react-router-dom';
import { render } from "@testing-library/react";
import CategorySelectPage from "./CategorySelectPage";

describe("CategorySelectPage tests", () => {
    it("renders correctly", () => {
        const { container } = render(
            <BrowserRouter>
                <CategorySelectPage />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });
});