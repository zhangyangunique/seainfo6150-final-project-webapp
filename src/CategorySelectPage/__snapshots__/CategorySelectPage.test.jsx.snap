// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySelectPage tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h2>
      Please select the topic you are interested in
    </h2>
    <div
      class="categoryCard"
    >
      <a
        href="/category/sport"
      >
        <p>
          Sport
        </p>
      </a>
      <a
        href="/category/food"
      >
        <p>
          Food
        </p>
      </a>
      <a
        href="/category/history"
      >
        <p>
          History
        </p>
      </a>
      <a
        href="/category/tourism"
      >
        <p>
          Tourism
        </p>
      </a>
    </div>
  </div>
</div>
`;
